### DISCUSSION TITLE
New - Filter box component

### DISCUSSION THREAD

**Discussion by Devaney** [Legal-and-General/canopy#878] [Canopy requests]
**Type of request**
New

**Component, pattern or foundation?**
Component

**Nature of request**
Requesting we create a new 'filter' component which consists of a filter button/toggle that gives the users access to a panel. 
![image](https://user-images.githubusercontent.com/7091410/169864165-50499db3-e832-4ede-ac6c-ff1c5275977e.png)

 In the designs created by Kevin, the panel show options for showing/hiding table columns.
  
![image](https://user-images.githubusercontent.com/7091410/169864317-eb6f1375-3338-4ae4-bbea-9cda8ed76196.png)

This will allow us to show five columns by default and allow the user to access other columns as they wish.

The mobile design for tables means that the filter would not be needed at mobile width.

**Comment by elenagarrone** [comment]
When the panel opens, does the page shift down or does the panel go over the content?

I can see the api of the component being something like:
```
<lg-filter-button iconName="columns">Lorem ipsum - it will automatically add the chevron icon</lg-filter-button>
<lg-filter-panel>
    <lg-filter-body>
        Here users will add the various filters
    </lg-filter-body>
    <lg-filter-footer>
        Here users will add the button group
    </lg-filter-footer>
</lg-filter-panel>
```

I would position the button outside so that other content can go on the side of it - like the design above.

@paul-request @owensgit @aaronBery what do you think?

**Reply by alexcanning** [reply]
When Kev discussed this component, it was with the idea of it pushing the content down.

**Comment by elenagarrone** [comment]
@Devaney go for it 👍

**Reply by elenagarrone** [reply]
Currently being worked on by the FEP team. I'll post an update once the PR is open.

**Comment by elenagarrone** [comment]
PR: https://github.com/Legal-and-General/canopy/pull/985

**Comment by elenagarrone** [comment]
Available in [v8.2.0](https://github.com/Legal-and-General/canopy/releases/tag/v8.2.0).
